sphinx/tutorial/rose/furthertopics/widget-development.rst

```diff
@@ -0,0 +1,156 @@
+Widget Development
+==================
+
+
+The ``rose edit`` GUI displays configurations using built-in widgets. For more
+complex requirements ``rose edit`` supports custom widgets as plugins.
+
+In this tutorial we will write a custom widget which offers typing suggestions
+when entering usernames.
+
+.. image:: img/rose-edit-custom-widget.png
+   :align: center
+   :width: 450px
+
+.. warning::
+
+   If you find yourself needing to write a custom widget, please contact the
+   Rose team for guidance.
```

Comment:
We could provide a 'mailto' link or hyperlink to our team page here?

Reply:
It could become a spam magnet. Will check how we want to give this information out.